components/layout_2020/flow/inline.rs

```diff
@@ -989,8 +989,10 @@ impl<'a, 'b> InlineFormattingContextState<'a, 'b> {
                 match self.current_line.count_justification_opportunities() {
                     0 => Length::zero(),
                     num_justification_opportunities => {
-                        (available_space - line_length) / (num_justification_opportunities as f32)
-                    },
+                        ((available_space - text_indent - line_length) /
+                            (num_justification_opportunities as f32))
+                            .max(Length::zero())
```

Comment:
Looks simpler like this:

```suggestion
                        (available_space - text_indent - line_length).max(Length::zero()) /
                            (num_justification_opportunities as f32)
```

Reply:
I've moved this to a new statement with a comment which, I think, is even clearer.